app/controllers/samples_controller.rb

```diff
@@ -49,6 +49,9 @@ class SamplesController < ApplicationController
   before_action only: :show_v2 do
     allowed_feature_required("report_v2")
   end
+  before_action only: :amr do
```

Comment:
Add an rspec test for user authorization.

Reply:
Added.